modules/node/handler/node_pagination_builder.e

```diff
@@ -0,0 +1,67 @@
+note
+	description: "Generic Node Pagination Builder Interface"
+	date: "$Date$"
+	revision: "$Revision$"
+
+deferred class
+	NODE_PAGINATION_BUILDER [G]
+
+feature -- Access
+
+	set_limit (a_limit: NATURAL)
+			-- Set limit with `a_limit'.
+		do
+			limit := a_limit
+		ensure
+			limit_set: limit = a_limit
+		end
+
+	set_offset (a_offset: NATURAL)
+			-- Set offset with `a_offset'.
+		do
+			offset := a_offset
+		ensure
+			limit_set: offset = a_offset
+		end
+
+	set_order_by_asc (a_field: READABLE_STRING_32)
```

Comment:
rename

set_ascending_order

and then
set_descending_order

otherwise it looks like it is dependent only on SQL queries